base/strings/basic.jl

```diff
@@ -610,9 +610,30 @@ julia> replace("abcdeγfgh", !isascii=>' ') # replace non-ASCII chars with space
 ```
 """
 isascii(c::Char) = bswap(reinterpret(UInt32, c)) < 0x80
-isascii(s::AbstractString) = all(isascii, s)
 isascii(c::AbstractChar) = UInt32(c) < 0x80
 
+function _isascii(bytes, first, last)
+    r = UInt8(0)
+    for n = first:last
+        @inbounds r |= bytes[n]
+    end
+    return r < 0x80
+end
+
+function isascii(s::AbstractString)
+    chunk_size = 1024
+    bytes = codeunits(s)
+    l = ncodeunits(s)
+    start = 1
+    fastmin(a,b) = ifelse(a < b, a, b)
+    while start <= l
+        @inline _isascii(bytes, start, fastmin(l, start + chunk_size)) || return false
```

Comment:
```suggestion
        @inline _isascii(bytes, start, fastmin(l, start+chunk_size-1)) || return false
```
Otherwise the length is `chunk_size+1`, which is probably not efficient. (Because of the minimum with `l` it won't lead to out-of-bounds access, though.)